core/iwasm/aot/aot_loader.c

```diff
@@ -2558,9 +2558,19 @@ load_function_section(const uint8 *buf, const uint8 *buf_end, AOTModule *module,
         }
     }
 
+#if defined(BUILD_TARGET_XTENSA)
+    /*
+     * For Xtensa XIP, real func_count is doubled, including aot_func and
+     * aot_func_internal, so need to divide func_count by 2 here.
+     */
+    if (module->is_indirect_mode)
+        size = sizeof(uint32) * (uint64)module->func_count / 2;
+    else
+#else
     size = sizeof(uint32) * (uint64)module->func_count;
+#endif
```

Comment:
If we don't emit double func_count in aot_emit_aot_file.c, then here we don't need to change. We can change the above reading func_ptrs and func_type_index to be like below?
```C
    size = sizeof(void *) * ((uint64)module->func_count * 2);
    if (size > 0
        && !(module->func_ptrs =
                 loader_malloc(size, error_buf, error_buf_size))) {
        return false;
    }
    for (i = 0; i < module->func_count * 2; i++) {
        ...
    }

    size = sizeof(uint32) * ((uint64)module->func_count * 2);
    if (size > 0
        && !(module->func_type_indexes =
                 loader_malloc(size, error_buf, error_buf_size))) {
        return false;
    }
    for (i = 0; i < module->func_count * 2; i++) {
        ...
    }
```

Reply:
hi @wenyongh ，module->func_count has been kept unchanged in latest commit 1fe0beacdbf53028fad5c1b271173af3a93e2104. 
Some macros and ifdef are added in aot_loader.c and aot_runtime.c like below:

```
uint32 func_count = module->func_count;
#if defined(BUILD_TARGET_XTENSA)
    /*
     * For Xtensa XIP, real func_count is doubled, including aot_func and
     * aot_func_internal, so need to multipy func_count by 2 here.
     */
    if (module->is_indirect_mode) {
        func_count *= 2;
    }
#endif

size = sizeof(void *) * ((uint64)module->func_count * 2);
    if (size > 0
        && !(module->func_ptrs =
                 loader_malloc(size, error_buf, error_buf_size))) {
        return false;
    }
    for (i = 0; i < module->func_count * 2; i++) {
        ...
    }
......
```